### PR TITLE
Add ASMHelper to decouple appserver code from ASM version opcode cons…

### PR DESCRIPTION
**Pull request by jgrassel**
Whenever ASM is updated, it often includes a version update in the form of Opcodes.ASM#, which code consuming ASM must update.  An update which is easy to overlook.

I've added `com.ibm.ws.asm.ASMHelper` (as part of Liberty's ASM bundle) to provide Liberty with a utility class to ease some of the pain:
```
    /**
     * Returns the constant representing the current version of ASM.
     *
     * @return
     */
    public final static int getCurrentASM();
```

I've found that certain instrumentation classes are too primordial (ant.instrumentation draws from `ant_build.js/build.gradle`, which uses the raw maven coordinates for asm, to benefit from this.  It may be possible to work around this, but that is for a future update.  For those, I went and updated the referenced ASM level to `Opcodes.ASM9'.

### CODE REVIEW

**Comment by jgrassel** [comment]
#build

**Comment by jgrassel** [comment]
#build

File: dev/com.ibm.ws.org.objectweb.asm/src/com/ibm/ws/asm/ASMHelper.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2010, 2013 IBM Corporation and others.
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.objectweb.asm/src/com/ibm/ws/asm/ASMHelper.java]: Copyright should be 2022

File: dev/com.ibm.ws.org.objectweb.asm/src/com/ibm/ws/asm/ASMHelper.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package com.ibm.ws.asm;
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.objectweb.asm/src/com/ibm/ws/asm/ASMHelper.java]: Ideally this package would be io.openliberty.asm

File: dev/com.ibm.ws.org.objectweb.asm/bnd.overrides [hunk 3]
```diff
@@ -8,4 +8,5 @@ Implementation-Title: ASM all classes
 DynamicImport-Package: *
```
Comment (jhanders34) [inline dev/com.ibm.ws.org.objectweb.asm/bnd.overrides:9]: A comment in bnd.bnd or bnd.overrides to instruct people to update the CURRENT_ASM field in ASMHelper if the version is changed to a new major version would be very helpful.  Somewhere near where the version is specified.

**Review by jhanders34 — CHANGES_REQUESTED** [review]

**Comment by jgrassel** [comment]
#run-libby-bot
#build

**Comment by jgrassel** [comment]
#run-libby-bot
#build

**Review by jhanders34 — APPROVED** [review]

**Pull request merged**